src/engine/IdTable.h

```diff
@@ -393,11 +436,17 @@ class IdTableImpl<0> {
     Row& operator*() { return _rowView; }
     const Row& operator*() const { return _rowView; }
 
+    pointer operator->() {return &_rowView;}
+    const value_type* operator->() const {return const_cast<const pointer>(&_rowView);}
```

Comment:
Why doesn't this just return `const pointer`? And do we need `const_cast<>` even though this does keep the pointer const?

Reply:
```
using pointer = int*;
const pointer x = nullptr;  // x has type int* const NOT const int*, must be initialized.
const int* y; // may alter the pointer, but not the pointee, is what we want here
```
I was confused myself.

And the const cast is not necessary, you are right. This was because of the initially wrong type. (And actually casts to the wrong type still, see above). Thanks for pointing it out.